### PR TITLE
Translate Plugin

### PR DESCRIPTION
**Pull request by YosiWeinreich**

### CODE REVIEW

**Comment by YosiWeinreich** [comment]
Translate

**Comment by YacovGold** [comment]
You profile picture is not updated...

File: Translate/TranslatePlugin.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace List
+{
+    record PersistentDataStructure(List<string> List);
+
+    public class TranslatePlugin : IPlugin
+    {
+        public const string _Id = "translate";
+        public string Id => _Id;
+
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs:17]: Extra enters

File: Translate/TranslatePlugin.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        public void Execute(PluginInput input)
+        {
+
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs]: Extra enter

File: Translate/TranslatePlugin.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                input.Callbacks.SavePluginUserData(str);
+
+                input.Callbacks.SendMessage($"the translation language: {str}");
+            }
+            else
+            {
+                input.Callbacks.SendMessage(translate(input.Message, input).ToString());
+            }
+        }
+        public string translate(string text, PluginInput input)
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs:43]: Here we need another enter

File: Translate/TranslatePlugin.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                { "target", len },
+                { "source", "" },
+            }),
+            };
+            using (var response = client.Send(request))
+            {
+                response.EnsureSuccessStatusCode();
+                var body = response.Content.ReadAsStringAsync();
+                return body.Result.Split(':')[3].Split(',')[0];
+
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs]: extra enter

File: Translate/TranslatePlugin.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                return "Type a language to translate:\"len \"";
+            }
+            var client = new HttpClient();
+            var request = new HttpRequestMessage
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri("https://google-translate1.p.rapidapi.com/language/translate/v2"),
+                Headers =
+            {
+                { "X-RapidAPI-Key", "de74d15ea7mshfaae56721a74a48p1edb42jsn4fa36d15cf91" },
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs]: Never a secret key should be in the code
Use environment variable instead, look here for example
https://github.com/YacovGold/chatbot/blob/develop/Runners/TelegramWebRunner/BotController.cs#L20

Also - invalidate the key

File: Translate/TranslatePlugin.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                { "X-RapidAPI-Host", "google-translate1.p.rapidapi.com" },
+            },
+                Content = new FormUrlEncodedContent(new Dictionary<string, string>
+            {
```
Comment (YacovGold) [inline Translate/TranslatePlugin.cs]: This should be indented

**Comment by YacovGold** [comment]
Also you should create a doc file to explain how to create a google secret key

**Pull request closed without merging**